README.md
Faience Next Generation Icon Theme
=================

Continued development of Faenza\Faience Icon Theme

Based on last official release 1.3.1

Make Faience Icons Great Again!

This work is licensed under the [GPL-3](https://gnu.org/licenses/gpl.html) license.



ChangeLog
***
===
* New actions icons: Inkscape, Gimp, etc.
* Change the folder structure.
* Faenza and Faience are merged
* Green folders theme
* More symbolic icons
1.3.1
===
* New weather symbolic icons
1.3
===
* New apps: UbuntuOne Music and Ubuntu Online Accounts
* New designs for Twitter, Spotify and adressbook
* Icons for system settings categories on Ubuntu
* Fix some missing links
1.2
===
* New apps: Desura, Gajim, Google Music Frame, Mail notification
* New icons sizes: 64x64 and 96x96
* Fixes the specific folders icons replaced by the standard one
* Fixes some missing links
1.1
===
* New apps: Gnome documents, Gnome contacts, Gnome online accounts, Gnome freecell, Clementine, Onboard, preferences-color, Screenruler, Scribes, Steam, Tracker, Wunderlist, Xterm
* New devices: tablet, network wired, network wireless, VPN, system harddisk
* Some reworked icons (Baobab, Gparted, Time-admin, Gconf editor, all folders)
* Emblems are more homogeneous
* Some new status icons and two new themes (Faenza-Ambiance and Feanza-Radiance) for a better integration with Unity in Ubuntu 11.10
1.0
===
* New apps: preferences-desktop-wallpaper, SEtroubleshoot, system-switch-java, Fedora autoplus, Fedora-release-notes
* New status icons: nm-device-wired-secure
* Some reworked icons
* A wide bunch of new symbolic icons for Gnome Shell
* Better support for XFCE
* All applications in 16x16
* Fix some missing links
0.9.2
=====
* New logos: Gentoo, Slackware
* New actions: folder-move, folder-copy
* New apps: libreoffice, workspace-switcher, wine notepad, winetricks, stellarium, mypaint
* New mimetypes: application-pgp-keys, encrypted
* New status icons: gsm 3G network, wireless secure network, bluetooth paired, weather icons in 24x24
* Color or opacity changes for some status icons
* A new design for Ubuntu One
* A few new 16x16 applications icons
0.9.1
=====
* A new Faenza-Darker theme with light monochrome icons for (dark) panels and menus and dark monochrome icons for toolbars
* Fix wrong links for system actions displayed in panel with Faenza-Darkest
* Fix missing links for XFCE panel
* Add gnome-fs-bookmark
* A few new 16x16 applications icons
0.9
===
* A new "darkest" theme with status and actions icons both in light grey
* A wide bunch of old and new devices and actions in all sizes
* New applications: debian software center, deja-dup, dc++, dvdrip, GCStar, guake, haguichi, Me TV, meld, file manager, nautilus actions configuration tool, tvtime, xine, zim
* New status icons for Cover Gloobus, Me TV, zim, guake, keepassx
* Highlight borders have been modified for all squared icons in 22x22 and 24x24
* All monochromes icons have been reworked for smoother contrast
* A new design for most of actions in 32x32 and above
* Some reworked icons for applications
* Subtle change in folders colour
* Some fixes

0.8
===
* New applications: adobe air, deadbeef, devede, devhelp, dia, facebook, flickr, frostwire, glade, gnucash, gnumeric, homebank, jdownloader, kupfer, netbeans, openbravo, openerp, openshot, phatch, picasa, qtcreator, radiotray, soundconverter, terminator, vim, wordpress, wxbanker, xbmc, xournal
* Reworked icons for gwibber, firefox, chromium and google chrome
* A whole set of emblems and weather icons
* Some new monochromatic icons for applications
* Some fixes for missing links and panels icons (especially with dark theme)
* More different sizes for existing icons

0.7
===
* New applications: desktop effects, easytag, emacs, emesene, flashplayer, gnome panel launcher, hardinfo, icedove, iceweasel, im-chooser, keepassx, last.fm, radio, mitter, monodevelop, openoffice template, openoffice printeradmin, pidgin, pino, pithos, podcast, firewall, sound juicer, truecrypt, wikipedia, wireshark, youtube
* New 16x16 icons for common applications
* A wide bunch of common mimetypes in all sizes
* New icons for treeviews, menu and toolbars in brasero, rhythmbox, banshee, synaptic and software center
* New monochrome status icons: dropbox, file copy, printer, getting things gnome, root privilege, pino.
* New logos for Archlinux and Frugalware.
* Some fixes for missing links and dark status icons
* Skype icons for Dockmanager
* Themes for Emesene

0.6
===
* New icons for applications: audacity, audacious, deluge, dropbox, eclipse, f-spot, google chrome, google earth, gmail, gtk-recordmydesktop, guayadeque, leafpad, midori, minitube, minitunes, pidgin, shotwell, skype, spotify, shutter, ubuntu-tweak, ubuntuone, wine, wine configuration, wine uninstaller, wine folder, zeitgeist.
* New icons for system preferences and administration: gparted, jockey, logviewer, network, sound, windows and shortcuts preferences, emerald, menu editor, gnome panel, opera widget manager, ATI catalyst control center.
* New icons for games: all standard Gnome games, defcon, gbrainy, openttd, world of goo
* New icons for devices: keyboard, scanner, floppy, pda.
* New bunch of monochromatic distributor logos for gnome main menu.
* Some fixes for dark status icons and wrong or missing icons names.

0.5
===
* First public release


app.php
<?php

// Sorry for ugly code

define('CURDIR', __DIR__);

if (isset($_SERVER['argv'][2])) {
  $file = $_SERVER['argv'][2] . '/' . str_replace('-symbolic', '', basename($_SERVER['argv'][1]));
} else {
  $file = CURDIR . '/' . str_replace('-symbolic', '', basename($_SERVER['argv'][1]));
}

$symbolic = $_SERVER['argv'][1];


//copy($_SERVER['argv'][1], $file);

$icon = basename($_SERVER['argv'][1], '-symbolic.svg');

$xml = simplexml_load_file(CURDIR . '/96x96b.svg');
$json = json_decode(file_get_contents('/run/media/guest/Private/clones/my-icons/Faenza/apps/apps.json'), true);

$gradient = $xml->xpath('//*[@id="buttonGradient"]')[0];
$gradient->stop[1]['stop-color'] = $json[$icon]['bg']['start'];
$gradient->stop[0]['stop-color'] = $json[$icon]['bg']['end'];

$sizes = explode(',', shell_exec("inkscape -S '$symbolic' | head -2 | tail -1"));
$sizes[2] = floatval($sizes[2]);
$sizes[4] = floatval($sizes[4]);

//file_put_contents($file, $xml->asXml());
$output = $xml->asXml();
$output = str_replace('</svg>', '', $output);
$output .= '<defs>
		<filter id="w">
			<feGaussianBlur stdDeviation="0.68880677"/>
		</filter>';

$output .= '
	<linearGradient id="a" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes[2] . '" x2="8" y2="' . ($sizes[2] + $sizes[4]) . '">
		<stop stop-color="' . $json[$icon]['fg']['start'] . '" offset="0"/>
		<stop stop-color="' . $json[$icon]['fg']['end'] . '" offset="1"/>
	</linearGradient>
   ';

$output .= '</defs>';

$xml = simplexml_load_file($symbolic);

  $path = clone($xml->path);
  $path['id'] = 'base';
	if (!isset($path['opacity'])) {
		$path['fill'] = 'url(#a)';
		//$path['opacity'] = '.80';
	} else {
		$path['fill'] = '#000';
		$path['opacity'] = '.30';
	}
  $path['fill'] = 'url(#a)';
  $output .= $path->asXML();

  if (count($xml->path) === 2) {
	$path = clone($xml->path[1]);
	if ((string) $path['fill'] === '#d40000' || (string) $path['fill'] === '#ff9000') {
		$path['fill'] = 'url(#b)';
	} else {
		$path['fill'] = '#000';
  		$path['opacity'] = '.30';
	}
	$output .= $path->asXML();
  }

  $output .= '<g id="Bevel">';
  $path = clone($xml->path);
  $path['opacity'] = '.40';
  $path['fill'] = '#000';
  $path['filter'] = 'url(#w)';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.40';
  $path['sodipodi:type'] = 'inkscape:offset';
  $path['inkscape:radius'] = '-1';
  $path['fill'] = '#000';
  //$path['filter'] = 'url(#w)';
  $path['inkscape:original'] = $path['d'];
  $path['transform'] = 'translate(0,1)';
  unset($path['d']);
  $output .= $path->asXML();
  $output .= '</g>';

  $output .= '<g id="BevelTop">';
  $path = clone($xml->path);
  $path['opacity'] = '.25';
  $path['sodipodi:type'] = 'inkscape:offset';
  $path['inkscape:radius'] = '0.5';
  $path['transform'] = 'translate(0,-1)';
  $path['fill'] = '#000';
  $path['inkscape:original'] = $path['d'];
  unset($path['d']);
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.25';
  $path['fill'] = '#000';
  $output .= $path->asXML();
  $output .= '</g>';

  $output .= '<g id="BevelShadow">';
  $path = clone($xml->path);
  $path['fill'] = '#000';
  $path['opacity'] = '.4';
  //$path['filter'] = 'url(#w)';
  $output .= $path->asXML();
  $output .= '</g>';

  $output .= '<g id="BevelHighlight">';
  $path = clone($xml->path);
  $path['opacity'] = '.2';
  $path['fill'] = '#fff';
  $path['transform'] = 'translate(0,2)';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.2';
  $path['fill'] = '#fff';
  $output .= $path->asXML();
  $output .= '</g>';


$output .= '</svg>';

file_put_contents($file, $output);
//
shell_exec("inkscape --file='$file' \
 --select=BevelHighlight --verb=SelectionUnGroup --verb=SelectionDiff --verb=EditDeselect \
 --select=Bevel --verb=SelectionUnGroup --verb=SelectionDiff \
 --select=BevelShadow --verb=SelectionUnGroup --verb=ObjectSetClipPath --verb=EditDeselect \
 --select=BevelTop --verb=SelectionUnGroup --verb=SelectionDiff --verb=EditDeselect \
 --verb=FileSave --verb=FileQuit");




big.php
<?php

// Sorry for ugly code

define('CURDIR', __DIR__);

if (isset($_SERVER['argv'][2])) {
  $file = CURDIR . '/' . str_replace('-symbolic', '', basename($_SERVER['argv'][1]));
} else {
  $file = CURDIR . '/' . str_replace('-symbolic', '', basename($_SERVER['argv'][1]));
}


copy($_SERVER['argv'][1], $file);

$xml = simplexml_load_file($file);

if (count($xml->path) === 1 && isset($xml->path['opacity'])) {
  if ($type === 1) {
    $xml->path['fill'] = '#000';
    $xml->path['opacity'] = '.30';
  } elseif ($type === 2) {
    $xml->path['fill'] = '#fff';
    $xml->path['opacity'] = '.40';
  }
  file_put_contents($file, $xml->asXML());
  //exit(0);
} else {

$sizes = explode(',', shell_exec("inkscape -S '$file' | head -2 | tail -1"));
$sizes[2] = floatval($sizes[2]);
$sizes[4] = floatval($sizes[4]);
/*
$output = '<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:xlink="http://www.w3.org/1999/xlink"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   height="' . $xml['height'] . '" width="' . $xml['width'] . '">
  <defs>
		<filter id="w">
			<feGaussianBlur stdDeviation="0.68880677"/>
		</filter>';
*/
$output = file_get_contents(CURDIR . '/96x96c.svg');
$output = str_replace('</svg>', '', $output);
$output .= '<defs>
		<filter id="w">
			<feGaussianBlur stdDeviation="0.68880677"/>
		</filter>';



  if ((string) $xml->path['fill'] === '#d40000') {
	  $output .= '
	<linearGradient id="a" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes[2] . '" x2="8" y2="' . ($sizes[2] + $sizes[4]) . '">
	  <stop stop-color="#ad0707" offset="0"/>
	  <stop stop-color="#f75535" offset="1"/>
	</linearGradient>
  ';
  } if ((string) $xml->path['fill'] === '#ff9000') {
	  $output .= '
	<linearGradient id="a" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes[2] . '" x2="8" y2="' . ($sizes[2] + $sizes[4]) . '">
	  <stop stop-color="#df880b" offset="0"/>
	  <stop stop-color="#f7c15a" offset="1"/>
	</linearGradient>
  ';
  } else {
  $output .= '
	<linearGradient id="a" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes[2] . '" x2="8" y2="' . ($sizes[2] + $sizes[4]) . '">
		<stop stop-color="#1e1e1e" offset="0"/>
		<stop stop-color="#505050" offset="1"/>
	</linearGradient>
   ';
  }

  if (count($xml->path) === 2) {
	$sizes2 = explode(',', shell_exec("inkscape -S '$file' | tail -1"));
	$sizes2[2] = floatval($sizes2[2]);
	$sizes2[4] = floatval($sizes2[4]);
	  $output .= '
	<linearGradient id="b" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes2[2] . '" x2="8" y2="' . ($sizes2[2] + $sizes2[4]) . '">
	  <stop stop-color="#ad0707" offset="0"/>
	  <stop stop-color="#f75535" offset="1"/>
	</linearGradient>
  ';
  }

  $output .= '</defs>';

  $path = clone($xml->path);
  $path['id'] = 'base';
	if (!isset($path['opacity'])) {
		$path['fill'] = 'url(#a)';
		$path['opacity'] = '.80';
	} else {
		$path['fill'] = '#000';
		$path['opacity'] = '.30';
	}
  $path['fill'] = 'url(#a)';
  $output .= $path->asXML();

  if (count($xml->path) === 2) {
	$path = clone($xml->path[1]);
	if ((string) $path['fill'] === '#d40000' || (string) $path['fill'] === '#ff9000') {
		$path['fill'] = 'url(#b)';
	} else {
		$path['fill'] = '#000';
  		$path['opacity'] = '.30';
	}
	$output .= $path->asXML();
  }

  $output .= '<g id="Bevel">';
  $path = clone($xml->path);
  $path['opacity'] = '.50';
  $path['fill'] = '#000';
  $path['filter'] = 'url(#w)';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.50';
  $path['sodipodi:type'] = 'inkscape:offset';
  $path['inkscape:radius'] = '-1';
  $path['fill'] = '#000';
  //$path['filter'] = 'url(#w)';
  $path['inkscape:original'] = $path['d'];
  $path['transform'] = 'translate(0,1)';
  unset($path['d']);
  $output .= $path->asXML();
  $output .= '</g>';

  $output .= '<g id="BevelShadow">';
  $path = clone($xml->path);
  $path['fill'] = '#000';
  $path['opacity'] = '.4';
  //$path['filter'] = 'url(#w)';
  $output .= $path->asXML();
  $output .= '</g>';

  $output .= '<g id="BevelHighlight">';
  $path = clone($xml->path);
  $path['opacity'] = '.6';
  $path['fill'] = '#fff';
  $path['transform'] = 'translate(0,1)';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.6';
  $path['fill'] = '#fff';
  $output .= $path->asXML();
  $output .= '</g>';


$output .= '</svg>';

file_put_contents($file, $output);
//
shell_exec("inkscape --file='$file' \
 --select=BevelHighlight --verb=SelectionUnGroup --verb=SelectionDiff --verb=EditDeselect \
 --select=Bevel --verb=SelectionUnGroup --verb=SelectionDiff \
 --select=BevelShadow --verb=SelectionUnGroup --verb=ObjectSetClipPath --verb=EditDeselect \
 --verb=FileSave --verb=FileQuit");
}

if (isset($_SERVER['argv'][2])) {
  copy($file, $_SERVER['argv'][2] . '/' . basename($file));
  unlink($file);
}


icon.php
<?php

// Sorry for ugly code

define('CURDIR', __DIR__);

if ($_SERVER['argv'][1] == 'Faience-ng') {
  $type = 3;
} elseif ($_SERVER['argv'][1] == 'Faience-ng-Dark') {
  $type = 2;
} else {
  $type = 1;
}

//$type = (int) $_SERVER['argv'][1];

if (isset($_SERVER['argv'][3])) {
  $file = CURDIR . '/' . str_replace('-symbolic', '', basename($_SERVER['argv'][2]));
} else {
  $file = CURDIR . '/' . str_replace('-symbolic', '', basename($_SERVER['argv'][2]));
}


copy($_SERVER['argv'][2], $file);

$xml = simplexml_load_file($file);

if (count($xml->path) === 1 && isset($xml->path['opacity'])) {
  if ($type === 1) {
    $xml->path['fill'] = '#000';
    $xml->path['opacity'] = '.30';
  } elseif ($type === 2) {
    $xml->path['fill'] = '#fff';
    $xml->path['opacity'] = '.40';
  } elseif ($type === 3) {
    $xml->path['fill'] = '#000';
    $xml->path['opacity'] = '.20';
  }
  file_put_contents($file, $xml->asXML());
  //exit(0);
} else {

$sizes = explode(',', shell_exec("inkscape -S '$file' | head -2 | tail -1"));
$sizes[2] = floatval($sizes[2]);
$sizes[4] = floatval($sizes[4]);

$output = '<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:xlink="http://www.w3.org/1999/xlink"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   height="' . $xml['height'] . '" width="' . $xml['width'] . '">
  <defs>';

if ($type === 1) {

  if ((string) $xml->path['fill'] === '#d40000') {
	  $output .= '
	<linearGradient id="a" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes[2] . '" x2="8" y2="' . ($sizes[2] + $sizes[4]) . '">
	  <stop stop-color="#ad0707" offset="0"/>
	  <stop stop-color="#f75535" offset="1"/>
	</linearGradient>
  ';
  } if ((string) $xml->path['fill'] === '#ff9000') {
	  $output .= '
	<linearGradient id="a" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes[2] . '" x2="8" y2="' . ($sizes[2] + $sizes[4]) . '">
	  <stop stop-color="#df880b" offset="0"/>
	  <stop stop-color="#f7c15a" offset="1"/>
	</linearGradient>
  ';
  } else {
  $output .= '
	<linearGradient id="a" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes[2] . '" x2="8" y2="' . ($sizes[2] + $sizes[4]) . '">
	 <stop stop-opacity=".86275" offset="0"/>
	 <stop stop-opacity=".47059" offset="1"/>
	</linearGradient>
   ';
  }

  if (count($xml->path) === 2) {
	$sizes2 = explode(',', shell_exec("inkscape -S '$file' | tail -1"));
	$sizes2[2] = floatval($sizes2[2]);
	$sizes2[4] = floatval($sizes2[4]);
	  $output .= '
	<linearGradient id="b" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes2[2] . '" x2="8" y2="' . ($sizes2[2] + $sizes2[4]) . '">
	  <stop stop-color="#ad0707" offset="0"/>
	  <stop stop-color="#f75535" offset="1"/>
	</linearGradient>
  ';
  }

  $output .= '</defs>';
  $output .= '<g id="Bevel">';
  $path = clone($xml->path);
  $path['opacity'] = '.15';
  $path['sodipodi:type'] = 'inkscape:offset';
  $path['inkscape:radius'] = '1';
  $path['fill'] = '#fff';
  $path['inkscape:original'] = $path['d'];
  unset($path['d']);
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.15';
  $path['fill'] = '#fff';
  $output .= $path->asXML();
  $output .= '</g>';

  $path = clone($xml->path);
  $path['id'] = 'base';
	if (!isset($path['opacity'])) {
		$path['fill'] = 'url(#a)';
	} else {
		$path['fill'] = '#000';
		$path['opacity'] = '.30';
	}
  $path['fill'] = 'url(#a)';
  $output .= $path->asXML();

  if (count($xml->path) === 2) {
	$path = clone($xml->path[1]);
	if ((string) $path['fill'] === '#d40000' || (string) $path['fill'] === '#ff9000') {
		$path['fill'] = 'url(#b)';
	} else {
		$path['fill'] = '#000';
  		$path['opacity'] = '.30';
	}
	$output .= $path->asXML();
  }

  $output .= '<g id="BevelShadow">';
  $path = clone($xml->path);
  $path['fill'] = '#000';
  $path['opacity'] = '.5';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.5';
  $path['fill'] = '#000';
  $path['transform'] = 'translate(0,1)';
  $output .= $path->asXML();
  $output .= '</g>';

  $output .= '<g id="BevelHighlight">';
  $path = clone($xml->path);
  $path['opacity'] = '.3';
  $path['fill'] = '#fff';
  $path['transform'] = 'translate(0,1)';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.3';
  $path['fill'] = '#fff';
  $output .= $path->asXML();
  $output .= '</g>';


} elseif ($type === 2) {
  if ((string) $xml->path['fill'] === '#d40000') {
	$output .= '
	<linearGradient id="a" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes[2] . '" x2="8" y2="' . ($sizes[2] + $sizes[4]) . '">
	  <stop stop-color="#f3604d" offset="0"/>
	  <stop stop-color="#c81700" offset="1"/>
	</linearGradient>
   ';
  } else {
	$output .= '
	  <linearGradient id="a" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes[2] . '" x2="8" y2="' . ($sizes[2] + $sizes[4]) . '">
		<stop stop-color="#ebebeb" offset="0"/>
		<stop stop-color="#aaa" offset="1"/>
	  </linearGradient>
	 ';
  }

  if (count($xml->path) === 2) {
	$sizes2 = explode(',', shell_exec("inkscape -S '$file' | tail -1"));
	$sizes2[2] = floatval($sizes2[2]);
	$sizes2[4] = floatval($sizes2[4]);
	$output .= '
	<linearGradient id="b" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes2[2] . '" x2="8" y2="' . ($sizes2[2] + $sizes2[4]) . '">
	  <stop stop-color="#f3604d" offset="0"/>
	  <stop stop-color="#c81700" offset="1"/>
	</linearGradient>
   ';
  }

  $output .= '</defs>';
  $output .= '<g id="Bevel">';
  $path = clone($xml->path);
  $path['opacity'] = '.25';
  ((string) $xml->path['fill'] !== '#bebebe') && $path['opacity'] = '.10';
  $path['fill'] = '#000';
  $path['inkscape:radius'] = '1';
  $path['inkscape:original'] = $path['d'];
  $path['sodipodi:type'] = 'inkscape:offset';
  unset($path['d']);
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.25';
  ((string) $xml->path['fill'] !== '#bebebe') && $path['opacity'] = '.10';
  $path['fill'] = '#000';
  $output .= $path->asXML();
  $output .= '</g>';

  $path = clone($xml->path);
  $path['id'] = 'base';
  $path['fill'] = 'url(#a)';
  $output .= $path->asXML();

  if (count($xml->path) === 2) {
	$path = clone($xml->path[1]);
	if ((string) $path['fill'] === '#d40000' || (string) $path['fill'] === '#ff9000') {
		$path['fill'] = 'url(#b)';
	} else {
		$path['fill'] = '#fff';
  		$path['opacity'] = '.40';
	}
	$output .= $path->asXML();
  }

  $output .= '<g id="BevelShadow">';
  $path = clone($xml->path);
  $path['fill'] = '#fff';
  $path['opacity'] = '.6';
  ((string) $xml->path['fill'] !== '#bebebe') && $path['opacity'] = '.30';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.6';
  ((string) $xml->path['fill'] !== '#bebebe') && $path['opacity'] = '.30';
  $path['fill'] = '#fff';
  $path['transform'] = 'translate(0,1)';
  $output .= $path->asXML();
  $output .= '</g>';

  $output .= '<g id="BevelHighlight">';
  $path = clone($xml->path);
  $path['opacity'] = '.45';
  ((string) $xml->path['fill'] !== '#bebebe') && $path['opacity'] = '.55';
  $path['fill'] = '#000';
  $path['transform'] = 'translate(0,1)';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.45';
  ((string) $xml->path['fill'] !== '#bebebe') && $path['opacity'] = '.55';
  $path['fill'] = '#000';
  $output .= $path->asXML();
  $output .= '</g>';

} elseif ($type === 3) {
  $l = round($sizes[2]) + 1;
  $k = 10 - $l - 1;
  if ($xml['height'] <= 16) {
	  $r = 11;
	  $cx = 9;
	  $cy = 10;
  } else {
	  $r = 16;
	  $cx = 12;
	  $cy = 13;
	  $l = round($sizes[2]) + 1;
	  $k = 11 - $l;
  }
  if ((string) $xml->path['fill'] === '#d40000') {
	$output .= '
	<radialGradient id="a" gradientUnits="userSpaceOnUse" cy="' . $cy . '" cx="' . $cx . '" gradientTransform="matrix(1 0 0 .' . $k . $l . $k . $l . $k . ' 0 ' . $l . '.' . $k . $l . $k . ($l+1) . ')" r="' . $r . '">
	  <stop stop-color="#e64b36" offset="0"/>
	  <stop stop-color="#a31414" offset="1"/>
	</radialGradient>
   ';
  } if ((string) $xml->path['fill'] === '#ff9000') {
	  $output .= '
	<linearGradient id="a" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes[2] . '" x2="8" y2="' . ($sizes[2] + $sizes[4]) . '">
	  <stop stop-color="#df880b" offset="0"/>
	  <stop stop-color="#f7c15a" offset="1"/>
	</linearGradient>
  ';
  } else {
  $output .= '
	<radialGradient id="a" gradientUnits="userSpaceOnUse" cy="' . $cy . '" cx="' . $cx . '" gradientTransform="matrix(1 0 0 .' . $k . $l . $k . $l . $k . ' 0 ' . $l . '.' . $k . $l . $k . ($l+1) . ')" r="' . $r . '">
	  <stop stop-opacity=".23529" offset="0"/>
	  <stop stop-opacity=".54902" offset="1"/>
	</radialGradient>
   ';
  }

  if (count($xml->path) === 2) {
	$sizes2 = explode(',', shell_exec("inkscape -S '$file' | tail -1"));
	$sizes2[2] = floatval($sizes2[2]);
	$sizes2[4] = floatval($sizes2[4]);
	  $output .= '
	<linearGradient id="b" gradientUnits="userSpaceOnUse" x1="8" y1="' . $sizes2[2] . '" x2="8" y2="' . ($sizes2[2] + $sizes2[4]) . '">
	  <stop stop-color="#c80000" offset="0"/>
	  <stop stop-color="#f3604d" offset="1"/>
	</linearGradient>
  ';
  }

  $output .= '</defs>';
  $output .= '<g id="Bevel">';
  $path = clone($xml->path);
  $path['opacity'] = '.05';
  $path['sodipodi:type'] = 'inkscape:offset';
  $path['inkscape:radius'] = '1';
  $path['fill'] = '#fff';
  $path['inkscape:original'] = $path['d'];
  unset($path['d']);
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.05';
  $path['fill'] = '#fff';
  $output .= $path->asXML();
  $output .= '</g>';

  $path = clone($xml->path);
  $path['id'] = 'base';
  $path['fill'] = 'url(#a)';
  $output .= $path->asXML();

  if (count($xml->path) === 2) {
	$path = clone($xml->path[1]);
	if ((string) $path['fill'] === '#d40000' || (string) $path['fill'] === '#ff9000') {
		$path['fill'] = 'url(#b)';
	} else {
		$path['fill'] = '#000';
  		$path['opacity'] = '.20';
	}
	$output .= $path->asXML();
  }

  //-----------
  $output .= '<g id="BevelInner">';

  $path = clone($xml->path);
  $path['opacity'] = '.3';
  $path['fill'] = '#000';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.3';
  $path['sodipodi:type'] = 'inkscape:offset';
  $path['inkscape:radius'] = '-0.5';
  $path['fill'] = '#000';
  $path['inkscape:original'] = $path['d'];
  unset($path['d']);
  $output .= $path->asXML();
  $output .= '</g>';

  //-----------
  $output .= '<g id="BevelShadow">';
  $path = clone($xml->path);
  $path['fill'] = '#000';
  $path['opacity'] = '.3';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.3';
  $path['fill'] = '#000';
  $path['transform'] = 'translate(0,1)';
  $output .= $path->asXML();
  $output .= '</g>';

  $output .= '<g id="BevelHighlight">';
  $path = clone($xml->path);
  $path['opacity'] = '.2';
  $path['fill'] = '#fff';
  $path['transform'] = 'translate(0,1)';
  $output .= $path->asXML();

  $path = clone($xml->path);
  $path['opacity'] = '.2';
  $path['fill'] = '#fff';
  $output .= $path->asXML();
  $output .= '</g>';

}

$output .= '</svg>';

//file_put_contents($file . '2', $output);
file_put_contents($file, $output);
//
shell_exec("inkscape --file='$file' --select=Bevel --verb=SelectionUnGroup --verb=SelectionDiff --verb=EditDeselect \
 --select=BevelShadow --verb=SelectionUnGroup --verb=SelectionDiff --verb=EditDeselect \
 --select=BevelHighlight --verb=SelectionUnGroup --verb=SelectionDiff --verb=EditDeselect \
 --select=BevelInner --verb=SelectionUnGroup --verb=SelectionDiff --verb=EditDeselect \
 --verb=FileSave --verb=FileQuit");
}

if (isset($_SERVER['argv'][3])) {
  copy($file, $_SERVER['argv'][3] . '/' . basename($file));
  unlink($file);
}


grub.php
<?php

// Sorry for ugly code

define('CURDIR', __DIR__);

$file = $_SERVER['argv'][1];

if (isset($_SERVER['argv'][2])) {
	$outfile = $_SERVER['argv'][2] . '/' . basename($file, '.svg') . '-symbolic.svg';
} else {
	$outfile = CURDIR . '/' . basename($file, '.svg') . '-symbolic.svg';
}



$xml = simplexml_load_file($file);
/*
$layers = $xml->xpath("//*[contains(@id,'layer') and @inkscape:label!='Symbol' and @inkscape:label!='Icon']");
foreach($layers as $layer) {
	unset($layer[0]);
}
*/

$paths = $xml->xpath("//*[contains(@style,'fill:url(#')]");

$paths2 = $xml->xpath("//*[(contains(@style,'opacity:0.3') or contains(@style,'opacity:0.2')) and contains(@style,'fill:#000000')]");

$size = (int) $xml['width'];
if ($size <= 16) {
	$size = 16;
} elseif($size <= 24) {
	$size = 22;
} else {
	$size = 96;
}

$output = '<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns="http://www.w3.org/2000/svg" height="' . $size . '" width="' . $size . '" version="1.1" xmlns:cc="http://creativecommons.org/ns#" xmlns:dc="http://purl.org/dc/elements/1.1/">
';

foreach ($paths as $path) {
	//unset($path['id']);
	//$path['fill'] = '#bebebe';
	//$output .= $path->asXML();
	if (isset($path['d']))
		$output .= '<path id="' . $path['id'] . '" d="' . $path['d'] . '" fill="#bebebe" />' . "\n";
}

foreach ($paths2 as $path) {
	//unset($path['id']);
	//$path['opacity'] = '.45';
	//$path['fill'] = '#bebebe';
	//$output .= $path->asXML();
	if (isset($path['d']))
		$output .= '<path id="' . $path['id'] . '" d="' . $path['d'] . '" fill="#bebebe" opacity=".45" />' . "\n";
}

$output .= "\n</svg>";

file_put_contents($outfile, $output);

$sizes = explode("\n", shell_exec("inkscape -S '$outfile'"));
unset($sizes[0]);
array_pop($sizes);

$xml = simplexml_load_file($outfile);

foreach($sizes as $s) {
	$arr = explode(',', $s);
	//var_dump($arr); exit(1);
	if (!isset($arr[1])) {
		continue;
	}
	if (intval($arr[1]) > $size || intval($arr[2]) > $size || intval($arr[2]) < 0 || intval($arr[1]) < 0) {
		list($element) = $xml->xpath('//*[@id="' . $arr[0] . '"]');
		unset($element[0]);
	}
}

file_put_contents($outfile, $xml->asXML());






